OneSignalSDK/onesignal/src/main/java/com/onesignal/WebViewManager.java

```diff
@@ -313,7 +313,7 @@ else if (!lastActivityName.equals(currentActivityName)) {
             // Navigate to new activity while displaying current IAM
             if (messageView != null)
                 messageView.removeAllViews();
-            showMessageView(lastPageHeight);
+            showMessageView(lastPageHeight); // removing this line prevented quick 2nd re-display in Jon's demo app
```

Comment:
Check out the git blame for this:
https://github.com/OneSignal/OneSignal-Android-SDK/blame/main/OneSignalSDK/onesignal/src/main/java/com/onesignal/WebViewManager.java#L312

It links to this PR
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1247

Which notes: "If user navigates with backpress while an IAM is being display, IAM should continue displaying until user dismiss it."

Can you test to see if this breaks this functionality?

Reply:
If the above does break then we probably need to check if the In-App Message is "dismissing" and omit trying to reshow it in this case. That is, the In-App Message is visible but is in the middle of playing it's dismiss animation.

Reply:
Yes removing that line does break the above-mentioned functionality.

Reply:
I found that _either_ **lowering** `ACTIVITY_FINISH_AFTER_DISMISS_DELAY_MS = 600` to something like 200 **_OR_** **raising** `ACTIVITY_INIT_DELAY = 200` to something like 600 seems to work. Don't quite understand the specifics of the functions that use these two variables nor the implications.

https://github.com/OneSignal/OneSignal-Android-SDK/blob/702634d4b3e083f1c0bbfc5c1797d3072bfd623e/OneSignalSDK/onesignal/src/main/java/com/onesignal/InAppMessageView.java#L56-L57